nice find. this code had a bit of a... stank ... to it?

build fail looks real.
```
======================================================================
ERROR: test_create (corehq.apps.api.tests.TestWebUserResource)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/api/tests.py", line 723, in test_create
    content_type='application/json')
  File "/mnt/commcare-hq-ro/corehq/apps/api/tests.py", line 200, in _assert_auth_post_resource
    response = self.client.post(api_url, post_data, content_type=content_type)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 482, in post
    secure=secure, **extra)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 296, in post
    secure=secure, **extra)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 358, in generic
    return self.request(**r)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 440, in request
    six.reraise(*exc_info)
  File "/vendor/lib/python2.7/site-packages/django/core/handlers/base.py", line 111, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/vendor/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 57, in wrapped_view
    return view_func(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 219, in wrapper
    response = callback(request, *args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 450, in dispatch_list
    return self.dispatch('list', request, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/api/resources/__init__.py", line 99, in dispatch
    return super(HqBaseResource, self).dispatch(request_type, request, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 482, in dispatch
    response = method(request, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 1390, in post_list
    updated_bundle = self.full_dehydrate(updated_bundle)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 895, in full_dehydrate
    data[field_name] = method(bundle)
  File "/mnt/commcare-hq-ro/corehq/apps/api/resources/v0_1.py", line 285, in dehydrate_role
    return bundle.obj.get_role(bundle.request.domain).name
AttributeError: 'NoneType' object has no attribute 'name'
```

--- a/corehq/apps/api/resources/v0_1.py
+++ b/corehq/apps/api/resources/v0_1.py
@@ -275,10 +275,12 @@ class WebUserResource(UserResource):
     permissions = fields.DictField()
 
     def dehydrate_role(self, bundle):
-        return bundle.obj.get_role(bundle.request.domain).name
+        role = bundle.obj.get_role(bundle.request.domain)
+        return role.name if role else ''
 
     def dehydrate_permissions(self, bundle):
-        return bundle.obj.get_role(bundle.request.domain).permissions._doc
+        role = bundle.obj.get_role(bundle.request.domain)
+        return role.permissions._doc if role else {}
 
     def dehydrate_is_admin(self, bundle):
         return bundle.obj.is_domain_admin(bundle.request.domain)

--- a/corehq/apps/users/models.py
+++ b/corehq/apps/users/models.py
@@ -574,7 +574,7 @@ class _AuthorizableMixin(IsMemberOfMixin):
     @memoized
     def has_permission(self, domain, permission, data=None):
         # is_admin is the same as having all the permissions set
-        if self.is_global_admin():
+        if (self.is_global_admin() and (domain is None or not domain_restricts_superusers(domain))):
             return True
         elif self.is_domain_admin(domain):
             return True
@@ -589,7 +589,6 @@ class _AuthorizableMixin(IsMemberOfMixin):
     def get_role(self, domain=None, checking_global_admin=True):
         """
         Get the role object for this user
-
         """
         if domain is None:
             # default to current_domain for django templates
@@ -600,7 +599,7 @@ class _AuthorizableMixin(IsMemberOfMixin):
 
         if checking_global_admin and self.is_global_admin():
             return AdminUserRole(domain=domain)
-        if self.is_member_of(domain): #need to have a way of seeing is_member_of
+        if self.is_member_of(domain):
             return self.get_domain_membership(domain).role
         else:
             raise DomainMembershipError()
@@ -1990,41 +1989,6 @@ class WebUser(CouchUser, MultiMembershipMixin, CommCareMobileContactMixin):
 
     def get_domains(self):
         return [dm.domain for dm in self.domain_memberships]
-
-    @memoized
-    def has_permission(self, domain, permission, data=None):
-        # is_admin is the same as having all the permissions set
-        if (self.is_global_admin() and (domain is None or not domain_restricts_superusers(domain))):
-            return True
-        elif self.is_domain_admin(domain):
-            return True
-
-        dm = self.get_domain_membership(domain)
-        if dm:
-            return dm.has_permission(permission, data)
-        else:
-            return False
-
-    @memoized
-    def get_role(self, domain=None, include_teams=True, checking_global_admin=True):
-        """
-        Get the role object for this user
-        """
-        if domain is None:
-            # default to current_domain for django templates
-            domain = self.current_domain
-
-        if checking_global_admin and self.is_global_admin():
-            return AdminUserRole(domain=domain)
-
-        if not include_teams:
-            return super(WebUser, self).get_role(domain)
-
-        dm = self.get_domain_membership(domain)
-        if not dm:
-            raise DomainMembershipError()
-
-        return dm.role
 
     @classmethod
     def get_admins_by_domain(cls, domain):

